src/saveload/newgrf_sl.cpp

```diff
@@ -91,11 +91,11 @@ struct NGRFChunkHandler : ChunkHandler {
 
 		int index = 0;
 
-		for (GRFConfig *c = _grfconfig; c != nullptr; c = c->next) {
+		for (const auto &c : _grfconfig) {
 			if (HasBit(c->flags, GCF_STATIC) || HasBit(c->flags, GCF_INIT_ONLY)) continue;
 			this->SaveParameters(*c);
 			SlSetArrayIndex(index++);
-			SlObject(c, description);
+			SlObject(&*c, description);
```

Comment:
Maybe, so it's consistent with `LoadCommon`.
```suggestion
			SlObject(c.get(), description);
```